cpp/Platform.Disposables/Disposable.h

```diff
@@ -22,22 +30,22 @@
 
         public: Disposable() { OnDispose = _emptyDelegate; }
 
-        public: Disposable(std::function<void()> action) : Disposable(action) { }
-
-        public: Disposable(std::function<Disposal> disposal) : Disposable(disposal) { }
 
         protected: void Dispose(bool manual, bool wasDisposed) override { this->RaiseOnDisposeEvent(manual, wasDisposed); }
 
         protected: void RaiseOnDisposeEvent(bool manual, bool wasDisposed) { this->OnDispose(manual, wasDisposed); }
 
         public: template <typename T> static bool TryDisposeAndResetToDefault(T* object)
         {
-            auto result = object.TryDispose();
+            auto result = object->TryDispose();
             if (result)
             {
                 object = 0;
```

Comment:
What's the point of nulling a local variable? I'm sure the compiler will even cut out this piece of dead code.

Reply:
In C#, it forces an object to be a candidate for garbage collection if no one else reference it.